plugins/k8sclient/api.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sclient
+
+import (
+	"github.com/ligato/networkservicemesh/plugins/idempotent"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// Interface is the interface to a k8sclient plugin
+type Interface interface {
```

Comment:
I would suggest calling this API rather than Interface.

I mostly have been using API for this sort of thing, I am not hugely religious on it, but we should at some point see about standardizing on *something* :)

Reply:
Makes sense to me.

Reply:
Well, k8s uses this convention of calling it `Interface` not sure which approach you decide to accept, just FYI

Reply:
I see other plugins we have in-tree use Interface. I don't really care either way, so I'll push a patch changing this to "API" soon unless @sbezverk and @edwarnicke resolve it the other way.